optuna_dashboard/ts/components/GraphRank.tsx

```diff
@@ -153,35 +146,67 @@ const getRankPlotInfo = (
   const xAxis = getAxisInfo(filteredTrials, xParam)
   const yAxis = getAxisInfo(filteredTrials, yParam)
 
-  const xValues: (string | number)[] = []
-  const yValues: (string | number)[] = []
+  let xValues: (string | number)[] = []
+  let yValues: (string | number)[] = []
   const zValues: number[] = []
   const hovertext: string[] = []
-  filteredTrials.forEach((trial) => {
-    const xValue =
-      trial.params.find((p) => p.name === xAxis.name)?.param_external_value ||
-      null
-    const yValue =
-      trial.params.find((p) => p.name === yAxis.name)?.param_external_value ||
-      null
-    if (trial.values === undefined || xValue === null || yValue === null) {
-      return
+  filteredTrials.forEach((trial, i) => {
+    if (xAxis.values[i] && yAxis.values[i] && trial.values) {
+      const xValue = xAxis.values[i] as string | number
+      const yValue = yAxis.values[i] as string | number
```

Comment:
How about avoiding the type assertion as follows?

```suggestion
    const xValue = xAxis.values[i]
    const yValue = yAxis.values[i]
    if (xValue && yValue && trial.values) {
```

Reply:
Fixed it.